fs/operations/operations.go

```diff
@@ -2165,3 +2166,28 @@ func GetFsInfo(f fs.Fs) *FsInfo {
 	}
 	return info
 }
+
+var interactiveMutex sync.Mutex
+
+func skipDesctructive(ctx context.Context, action string) (string, bool) {
```

Comment:
Destructive is spelled wrong here :-)

I think this function needs to take an action (eg `delete`, `move`, `copy`) and a subject (eg  `fs.LogDirName(f, dir)`)

I can see in a future commit (or maybe this one) we'll want to be able to give the user a menu something like this

```
About to ACTION SUBJECT
y) Yes ACTION 
n) No, do not ACTION (default)
s) Skip all future ACTION with no more questions
a) Allow all future ACTION with no more questions
```

So I think the function should look like

```
func skipDestructive(action, subject string) bool {
}
```

Reply:
Could you please point me, how to code `skip all future` feature? I think only way in write to config at runtime. 

I added `ctx` here in the arguments because want to count a objects\dirs to be processed to show in the log message.

Reply:
It would just be `skip all future` in this run of rclone.  So I'd make a map which is protected by your mutex

```
var (
    interactiveMutex sync.Mutex
    interactiveSkip = map[string]bool{}
    interactiveAllow = map[string]bool{}
```

Then see if `interactiveSkip[action]` or `interactiveAllow[action]` was set to immediately return `false` or `true`. When the user sets the option you set `interactiveSkip[action] = true`.

> I added ctx here in the arguments because want to count a objects\dirs to be processed to show in the log message.

I don't understand what you mean here? Anyway adding `ctx` is probably a good idea!